mriqc/viz/fmriplots.py

```diff
@@ -135,24 +135,34 @@ def fmricarpetplot(func_data, segmentation, outer_gs, tr=None, nskip=0):
 
     # Carpet plot
     ax1 = plt.subplot(gs[1])
-    ax1.imshow(detrended[order, :], interpolation='nearest',
+
+    # Avoid segmentation faults for long acquisitions by decimating the input data
+    long_cutoff = 800
+    if detrended.shape[1] > long_cutoff:
+        data = detrended[order, ::2]
+    else:
+        data = detrended[order, :]
+
+    ax1.imshow(data, interpolation='nearest',
```

Comment:
I think 800 points is not a very large axis for an image in matplotlib. The problem is that the generated file is extremely large, right?. We should try first without decimating, but using the imshow ```extent``` argument. I'd say that fixing that then the resulting axis will not explode. WDYT?

We can merge this as a first approximation, but I will need to look closer at this. Seems to me little adaptive.